core/src/main/java/org/mule/component/DefaultComponentLifecycleAdapter.java

```diff
@@ -339,7 +338,7 @@ public Object invoke(MuleEvent event) throws MuleException
         {
             if (componentObject == null)
             {
-                throw new ComponentException(createStaticMessage("componentObject is null"), RequestContext.getEvent(), component);
+                throw new ComponentException(createStaticMessage("componentObject is null"), event, component);
```

Comment:
Didn't work. Because when the ComponentException is created with a reference to the "wrong event" (the one created by the dispatch process) this event is later loaded in RequestContext before calling the component defined in the catch-exception-strategy block using that reference set in ComponentException and later stored in RequestContext.